addons/web/static/src/views/graph/graph_renderer.js

```diff
@@ -272,6 +373,36 @@ export class GraphRenderer extends Component {
         return { data, options, type: mode };
     }
 
+    /**
+     * Returns the animation options.
+     * 1. This adds progressive animation for Bar & Line charts.
+     * 2. Reduce animation duration for Pie chart.
+     * @returns {Object}
+     */
+    getAnimationOptions() {
+        let delayed;
+        const { mode } = this.model.metaData;
+        const labelsCount = this.model.data.labels.length;
+        const gap = 350;
+        const animationOptions = {};
+        if(mode != 'pie') {
```

Comment:
again. It is better to inverse conditions
if (mode === "pie") {
...
} else {
...
}